### DISCUSSION TITLE
Increased(?) Memory usage after 10.5.16

### DISCUSSION THREAD

**Discussion by rliebi** [pimcore/pimcore#14510] [Q&A / Support] [answered]
Hi
We ran into a problem recently. Since Pimcore v10.5.16 we are getting memory allocation errors when opening the data object class browser, a user object or certain objects types. With v10.5.15 we did not encounter this issue.

```
Fatal error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in /var/www/html/vendor/pimcore/pimcore/lib/Cache/RuntimeCache.php on line 97

Fatal error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in /var/www/html/vendor/symfony/var-dumper/Cloner/AbstractCloner.php on line 240
```

I don't know anymore where to search, the XDebug profiler says such a request takes about 40MB of memory.

If I downgrade to 10.5.15 again all works fine. I could not determine a change between versions which could cause an issue.

Does anyone have an idea where to look?

**Comment by milovangibaca** [comment]
Hello,
We have similar problems with memory allocation.
In our case, it happens on v10.5.15 and after upgrading to the latest version 10.5.18 the issue is still there:

`FastCGI sent in stderr: "PHP message: PHP Fatal error:  Allowed memory size of 1073741824 bytes exhausted (tried to allocate 20480 bytes) in /local/htdocs/project/vendor/myclabs/deep-copy/src/DeepCopy/Reflection/ReflectionHelper.php on line 38PHP message: PHP Fatal error:  Allowed memory size of 1073741824 bytes exhausted (tried to allocate 65536 bytes) in /local/htdocs/project/vendor/composer/ClassLoader.php on line 571" while reading response header from upstream, client: 67.34.151.232, server: project.com, request: "GET /admin/object/get?_dc=1677874966358&id=323 HTTP/1.1", upstream: "fastcgi://unix:/var/run/php/www.sock:", host: " project.com", referrer: "https://project.com/admin/?_dc=1677873280&perspective="`

**Comment by NiklasBr** [comment]
Thought we were the only ones. Can reproduce the following memory issue which has appeared recently and we are not sure why exactly, but it seems related to DataObject Listings or Paginations.

```
Fatal error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 212992 bytes) in /var/pim/cache/prod/twig/78/78fb2ffcad74e18a9a709e1be863a14c.php on line 70

Fatal error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in /var/www/pimcore/vendor/symfony/cache/Marshaller/DefaultMarshaller.php on line 80
```

**Comment by qmaxquique** [comment]
Hello, I am experiencing the same issue.  We raised the memory limit for PHP up to 1.5G and it continues to happen.
The error log points to the file
```
/vendor/myclabs/deep-copy/src/DeepCopy/Reflection/ReflectionHelper.php on line 38
```

**Comment by NiklasBr** [comment]
Found the issue in another environment today, same `ReflectionHelper` in the error message as other have seen:

```
Fatal error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in
/var/www/pimcore/vendor/pimcore/pimcore/models/DataObject/Concrete.php on line 525

Fatal error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in
/var/www/pimcore/vendor/myclabs/deep-copy/src/DeepCopy/Reflection/ReflectionHelper.php on line 38
```

**Reply by milovangibaca** [reply]
@NiklasBr,
Do you have "Lazy Loading" enabled in your class definition? 
I disabled it and I think that solved my problem, but I'm still testing it, not sure 100%.

**Reply by NiklasBr** [reply]
@milovangibaca Yes, on multiple fields in the class definition.

**Comment by podarcis** [comment]
You might want to try to narrow it down using `git bisect` between 10.5.15 and 10.5.16 in a cloned environment.

**Comment by rliebi** [comment]
thanks to all for your suggestions. It seems 10.5.19 improves the behaviour for us. it works not with 512M memory_limit but starting from 540M it seems to work. we already increased our memory_limit on our QA servers to 768M so we will try it out.

we will also check git bisect if we got some available resources.

**Reply by rliebi** [reply]
unfortunately it only improves things locally. our first stage server crashes even with memory_limit=-1
will need to investigate further.

**Comment by rliebi** [comment, marked as answer]
We figured it out. Finally. 
We have an OptionsProvider which depends on loading configuration entries from `configurationRepository` or loading `ClassificationKeyConfig`. This resulted somehow in an endless loop inside of the Caching Engine as it crashed during loading the ClassDefinition file.
Solution is inside of the SelectOptionsProviderInterface::getOptions add the following Check:
```
if ($context['object'] === null) {
            return [];
}
```
